File: Backend/Api/Core/Models/Habitacion.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;

namespace Api.Core.Models
{
    public class Habitacion : EntidadConId
    {
        [Required, MaxLength(12)]
        public string Nombre { get; set; }

        [Required]
        public bool TieneBanio { get; set; }

        [Required]
        public bool EsPrivada { get; set; }

        [MaxLength(140)]
        public string InformacionAdicional { get; set; }

        public ICollection<CamaIndividual> CamasIndividuales { get; set; }
        public ICollection<CamaMatrimonial> CamasMatrimoniales { get; set; }
        public ICollection<CamaCucheta> CamasCuchetas { get; set; }

        public int LugaresLibresEntre(DateTime desde, DateTime hasta)
        {
            var sumaIndividuales = CamasIndividuales?.Sum(x => x.LugaresLibresEntre(desde, hasta)) ?? 0;
            var sumaMatri = CamasMatrimoniales?.Sum(x => x.LugaresLibresEntre(desde, hasta)) ?? 0;
            var sumaCucheta = CamasCuchetas?.Sum(x => x.Abajo.LugaresLibresEntre(desde, hasta) + x.Arriba.LugaresLibresEntre(desde, hasta)) ?? 0;

            return sumaIndividuales + sumaCucheta + sumaMatri;
        }
    }
}

File: Frontend/src/interfaces/habitacion.ts
export enum CamaTipo {
  Individual = 'Individual',
  Matrimonial = 'Matrimonial',
  CuchetaArriba = 'Cucheta Arriba',
  CuchetaAbajo = 'Cucheta Abajo',
}
export interface CamaDTO {
  id: number;
  nombre: string;
  tipo: CamaTipo;
}

interface CamaCuchetaDTO {
  id: number;
  nombre: string;
  abajo: CamaDTO;
  arriba: CamaDTO;
}

export interface HabitacionDTO {
  id: number;
  nombre: string;
  tieneBanio: boolean;
  esPrivada: boolean;
  informacionAdicional: string;
  camasIndividuales: CamaDTO[];
  camasCuchetas: CamaCuchetaDTO[];
  camasMatrimoniales: CamaDTO[];
}
